tests/test_controllers.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+import pytest
+import torch as th
+
+import omnigibson as og
+import omnigibson.utils.transform_utils as T
+from omnigibson.robots import LocomotionRobot
+
+
+def test_arm_control():
+    # Create env
+    cfg = {
+        "scene": {
+            "type": "Scene",
+        },
+        "objects": [],
+        "robots": [
+            {
+                "type": "FrankaPanda",
+                "obs_modalities": "rgb",
+                "position": [150, 150, 100],
+                "orientation": [0, 0, 0, 1],
+            },
+            {
+                "type": "Fetch",
+                "obs_modalities": "rgb",
+                "position": [150, 150, 105],
+                "orientation": [0, 0, 0, 1],
+            },
+            {
+                "type": "Tiago",
+                "obs_modalities": "rgb",
+                "position": [150, 150, 110],
+                "orientation": [0, 0, 0, 1],
+            },
+        ],
+    }
+    env = og.Environment(configs=cfg)
+
+    # Define error functions to use
+    def check_zero_error(curr_position, init_position, tol=1e-2):
+        return th.norm(curr_pos - init_pos).item() < tol
+
+    def check_forward_error(curr_position, init_position, tol=1e-2, forward_tol=1e-2):
+        # x should be positive
+        return (curr_position[0] - init_position[0]).item() > forward_tol and th.norm(
+            curr_position[[1, 2]] - init_position[[1, 2]]
+        ).item() < tol
+
+    def check_side_error(curr_position, init_position, tol=1e-2, side_tol=1e-2):
+        # y should be positive
+        return (curr_position[1] - init_position[1]).item() > side_tol and th.norm(
+            curr_position[[0, 2]] - init_position[[0, 2]]
+        ).item() < tol
+
+    def check_up_error(curr_position, init_position, tol=1e-2, up_tol=1e-2):
+        # z should be positive
+        return (curr_position[2] - init_position[2]).item() > up_tol and th.norm(
+            curr_position[[0, 1]] - init_position[[0, 1]]
+        ).item() < tol
```

Comment:
Is it possible to do a quantitative check, rather than a qualitative test?
e.g. command the eef to move 5cm and confirm it is the case.

Reply:
done!